### DISCUSSION TITLE
Alpine 3.x.x x-for does not support nested data structures

### DISCUSSION THREAD

**Discussion by practiv-wamci** [alpinejs/alpine#3923] [7. Bugs (archived)] [answered]
Here's a really simple example of the problem:

```
<html>
  <head>
    <script
      defer
      src="https://cdnjs.cloudflare.com/ajax/libs/alpinejs/3.13.3/cdn.js"
    ></script>
  </head>
  <body>
    <div
      x-data="{ list: [
          {name: 'Alice', tasks: ['task1', 'task2']},
          {name: 'Bob', tasks: ['task3', 'task4']}
      ]}"
    >
      <template x-for="(item, index) in list" :key="index">
        <h2 x-text="item.name"></h2>
        <ul>
          <template x-for="(task, taskIndex) in item.tasks" :key="taskIndex">
            <li x-text="task"></li>
          </template>
        </ul>
      </template>
    </div>
  </body>
</html>
```

Seems like this it not the first time Alpinejs is faced with this issue?

**Comment by practiv-wamci** [comment]
Also, I'd like to be able to work with nested objact data structures, ie. 
`{name: 'Bob', tasks: [{ id: 3, desc: 'task3'}, { id: 4, desc: 'task4'} ]}`

and this access it using:
`<li x-text="task.desc"></li>`

**Comment by SimoTod** [comment, marked as answer]
You can only one root element inside your template. See https://alpinejs.dev/directives/for

**Comment by ekwoka** [comment]
It does. What it does not support is multiple direct children inside a template element.
